backend/routes/addUser.js

```diff
@@ -27,10 +27,24 @@ router.post('/api/add-users', async (req, res) => {
     if (!user) {
         return res.status(401).json({ error: 'unauthorized' });
     }
-    if (!(await checkPassword(user.userId, hash(password)))) {
-        return res.status(401).json({ error: 'invalid password' });
+    try {
+        if (!(await checkPassword(user.userId, hash(password)))) {
+            return res.status(401).json({ error: 'invalid password' });
+        }
+    }
+    catch(error){
+        console.log(error);
+    }
```

Comment:
If the `checkPassword` fails, password check is completely skipped since `catch` block just falls through. There should be some proper status 500 response and `return`.